pkg/cmd/server/api/types.go

```diff
@@ -144,6 +144,24 @@ type ProjectConfig struct {
 	// It is in the format namespace/template and it is optional.
 	// If it is not specified, a default template is used.
 	ProjectRequestTemplate string
+
+	// SecurityAllocator controls the automatic allocation of UIDs and MCS labels to a project. If nil, allocation is disabled.
+	SecurityAllocator *SecurityAllocator
+}
+
+type SecurityAllocator struct {
+	// UIDAllocatorRange defines the total set of Unix user IDs (UIDs) that will be allocated to projects automatically, and the size of the
+	// block each namespace gets. For example, 1000-1999/10 will allocate ten UIDs per namespace, and will be able to allocate up to 100 blocks
+	// before running out of space. The default is to allocate from 1 billion to 2 billion in 10k blocks (which is the expected size of the
+	// ranges Docker images will use once user namespaces are started).
+	UIDAllocatorRange string
+	// MCSAllocatorRange defines the range of MCS categories that will be assigned to namespaces. The format is
+	// "<prefix>/<numberOfLabels>[,<maxCategory>]". The default is "s0/2" and will allocate from c0 -> c1023, which means a total of 535k labels
```

Comment:
s0 is the prefix and defines levels.  SELinux already uses colons and commas, so we need a separator in the range, which is what / is.  See the original pull for more details.

> On Jun 8, 2015, at 8:08 AM, David Eads <notifications@github.com> wrote:
> 
> In pkg/cmd/server/api/types.go:
> 
> > @@ -144,6 +144,24 @@ type ProjectConfig struct {
> >  	// It is in the format namespace/template and it is optional.
> >  	// If it is not specified, a default template is used.
> >  	ProjectRequestTemplate string
> > +
> > +	// SecurityAllocator controls the automatic allocation of UIDs and MCS labels to a project. If nil, allocation is disabled.
> > +	SecurityAllocator *SecurityAllocator
> > +}
> > +
> > +type SecurityAllocator struct {
> > +	// UIDAllocatorRange defines the total set of Unix user IDs (UIDs) that will be allocated to projects automatically, and the size of the
> > +	// block each namespace gets. For example, 1000-1999/10 will allocate ten UIDs per namespace, and will be able to allocate up to 100 blocks
> > +	// before running out of space. The default is to allocate from 1 billion to 2 billion in 10k blocks (which is the expected size of the
> > +	// ranges Docker images will use once user namespaces are started).
> > +	UIDAllocatorRange string
> > +	// MCSAllocatorRange defines the range of MCS categories that will be assigned to namespaces. The format is
> > +	// "<prefix>/<numberOfLabels>[,<maxCategory>]". The default is "s0/2" and will allocate from c0 -> c1023, which means a total of 535k labels
> I'm not knowledgeable in the art, but I think it's really confusing that "s0/2" results in something starting with "c". Is that normal or a typo?
> 
> —
> Reply to this email directly or view it on GitHub.
>